### PR TITLE
#166283339 add books based on category

### PR DESCRIPTION
**Pull request by PelumiAlesh**
### What does this PR do?
Add categories table to the database and let a user add books based on category
#### How should this be manually tested?
After cloning the repo , follow this steps:

- Run the following commands
        `npm i`
        `npm run migrate:undo`
        `npm run migrate`
        `npm run seed`

- Make a `POST` request to this endpoint `/api/v1/books`

#### What are the relevant pivotal tracker stories?
[#166283339](https://www.pivotaltracker.com/story/show/166283339 "A user should be able to add books based on category or genre")

#### Screenshots (if appropriate)
Screenshot 2019-06-07 at 2 39 50 PM
#### Questions:

### CODE REVIEW

File: server/__test__/mock_data/mock_categories.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const categories = {
```
Comment (OvieMudi) [inline server/__test__/mock_data/mock_categories.js:1]: Cool. We could also use a "Science Fiction" category. That's my fav.

File: server/db/migrations/20190606195726-create-book.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ export function up(queryInterface, Sequelize) {
     id: {
       allowNull: false,
       autoIncrement: true,
-      primaryKey: true,
+      primarykey: true,
```
Comment (emp-daisy) [inline server/db/migrations/20190606195726-create-book.js]: Confirm this is right?

File: server/helpers/utilities.js [hunk 3]
```diff
@@ -35,15 +35,21 @@ class Utilities {
...
+  * @param {integer} key - the table's id to be checked against in database
+  * @param {table} model - the table name
+  * @returns {boolean / dataValue} returns dataValue or false
   * @memberof Utilities
   */
 
-  static async checkAuthorID(authorID) {
-    const author = await models.Authors.findOne( { where: { id : authorID} });
-    return author;
+  static async exits(key, model) {
```
Comment (emp-daisy) [inline server/helpers/utilities.js:45]: Nice work with the generic function

File: server/db/migrations/20190606195726-create-book.js (hunk 2 above)
Comment (PelumiAlesh) [reply server/db/migrations/20190606195726-create-book.js]: Yes, it is right.

File: server/helpers/utilities.js (hunk 3 above)
Comment (PelumiAlesh) [reply server/helpers/utilities.js:45]: Thanks

File: server/db/migrations/20190605110045-create-categories.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable require-jsdoc */
+
+export function up(queryInterface, Sequelize) {
+  return queryInterface.createTable('Categories', {
+    id: {
+      allowNull: false,
+      autoIncrement: true,
+      primaryKey: true,
```
Comment (emp-daisy) [inline server/db/migrations/20190605110045-create-categories.js:8]: If the lowercase is right, what about here?

**Review by emp-daisy — APPROVED** [review]

**Review by OvieMudi — APPROVED** [review]

**Review by Hadeneekeh — APPROVED** [review]

**Pull request merged**